src/sentry/static/sentry/app/index.js

```diff
@@ -1,15 +1,47 @@
 import '@babel/polyfill';
 
-// Initial data hydration. The __initialData will be set by the django backend
-// serving sentry.
-if (window.__initialData) {
-  const {distPrefix, csrfCookieName, sentryConfig, userIdentity} = window.__initialData;
+const BOOTSTRAP_URL = '/api/client-config/';
 
+const bootApplication = data => {
+  const {distPrefix, csrfCookieName, sentryConfig, userIdentity} = data;
+
+  // TODO(epurkhiser): Would be great if we could remove some of these from
+  // existing on the window object and instead pass into a bootstrap function.
+  // We can't currently do this due to some of these globals needing to be
+  // available for modules imported by the bootstrap.
   window.csrfCookieName = csrfCookieName;
   window.__sentryGlobalStaticPrefix = distPrefix;
   window.__SENTRY__OPTIONS = sentryConfig;
   window.__SENTRY__USER = userIdentity;
+  window.__initialData = data;
+
+  // Once data hydration is done we can initialize the app
+  require('./bootstrap');
+};
+
+async function bootWithHydration() {
+  const response = await fetch(BOOTSTRAP_URL);
```

Comment:
We should probably show an error when this fails.

Reply:
Agreed. I'm not super worried about it right now, since this isn't something that will be used anywhere but frontend previews at the moment though.